networkx/classes/function.py

```diff
@@ -186,16 +191,9 @@ def freeze(G):
     --------
     is_frozen
     """
-    G.add_node=frozen
-    G.add_nodes_from=frozen
-    G.remove_node=frozen
-    G.remove_nodes_from=frozen
-    G.add_edge=frozen
-    G.add_edges_from=frozen
-    G.remove_edge=frozen
-    G.remove_edges_from=frozen
-    G.clear=frozen
-    G.frozen=True
+    for method in MUTATORS:
+        setattr(G, method, frozen)
```

Comment:
Seems unnecessary to me. The attributes are not "random", nor are they arbitrary. They are fixed, and if any of the method names change, unit tests will catch such an error.